src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -406,6 +417,11 @@ void WiFiManager::ConnectHandler(uint8_t * data)
         if (Instance().mHandling.mOnConnectionSuccess)
         {
             Instance().mHandling.mOnConnectionSuccess();
+#if CONFIG_CHIP_WIFI_CONNECTION_RECOVERY
+            // Reset the connection recover interval to the minimum value after the defined delay
+            DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(kConnectionRecoveryDelayToReset), ResetRecoveryTime,
```

Comment:
Cant we just call `ResetRecoveryTime` immediately? I mean, the only thing this method does is changing the current recovery time. I guess this member is used in one thread only anyway (if not, we would need some synchronization primitives), so it won't affect any ongoing operation.

Reply:
I guess we should add a possibility to delay resetting the recovery time to avoid frequent reconnection when the connection has poor link quality.